Applications/DataExplorer/DataView/DiagramView/DiagramList.cpp

```diff
@@ -288,19 +283,38 @@ int DiagramList::readList(const SensorData* data, std::vector<DiagramList*> &lis
     return nLists;
 }
 
-void DiagramList::setList(std::vector< std::pair<QDateTime, float> > coords)
+void DiagramList::truncateToRange(QDateTime const& start, QDateTime const& end)
 {
-    int numberOfDays;
+    float start_secs = static_cast<float>(_startDate.secsTo(start));
+    if (start_secs < 0)
+        start_secs = 0;
+    float end_secs = static_cast<float>(_startDate.secsTo(end));
+    if (end_secs < start_secs)
+        end_secs = _coords.back().first;
+
+    if (start_secs == 0 && end_secs == _coords.back().first)
+        return;
+
+    _coords.erase(std::remove_if(_coords.begin(), _coords.end(),
+        [&](std::pair<float, float> const& c){return (c.first<start_secs || c.first>end_secs);}),
+        _coords.end());
+    _startDate = start;
+    for (auto& c : _coords)
+        c.first -= start_secs;
+    update();
+}
 
+void DiagramList::setList(std::vector< std::pair<QDateTime, float> > coords)
```

Comment:
Do you really want to copy the vector? Maybe `const &`?

Reply:
done :heavy_check_mark: